org.eclipse.jdt.ls.tests/src/org/eclipse/jdt/ls/core/internal/managers/MavenProjectImporterTest.java

```diff
@@ -132,6 +134,26 @@ public void testInvalidProject() throws Exception {
 		assertIsMavenProject(invalid);
 	}
 
+	@Test
+	public void testDeleteClasspath() throws Exception {
+		attachJobSpy();
+		String name = "salut";
+		importProjects("maven/" + name);
+		IProject project = getProject(name);
+		assertIsJavaProject(project);
+		assertIsMavenProject(project);
+		IFile dotClasspath = project.getFile(IJavaProject.CLASSPATH_FILE_NAME);
+		File file = dotClasspath.getRawLocation().toFile();
+		assertTrue(file.exists());
+		file.delete();
+		projectsManager.fileChanged(file.toPath().toUri().toString(), CHANGE_TYPE.DELETED);
+		importExistingMavenProject(name);
```

Comment:
should check .classpath has been recreated

Reply:
Fixed.

Reply:
works for maven, but doesn't work for gradle

Reply:
sorry i was testing on a non-java gradle project. What doesn't seem to work is .classpath regeneration on workspace reopening